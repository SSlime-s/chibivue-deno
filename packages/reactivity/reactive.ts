import { mutableHandlers } from "./baseHandler.ts";

export function reactive<T extends object>(target: T): T {
  const proxy = new Proxy(target, mutableHandlers);

  return proxy as T;
}
