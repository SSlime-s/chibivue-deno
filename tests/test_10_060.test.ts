import { createApp } from "chibivue";
import { HOST_ID, setup } from "./setup.ts";
import { assertSnapshot } from "@std/testing/snapshot";
import { reactive } from "../packages/reactivity/reactive.ts";

Deno.test("template もコンパイルできる", async (t) => {
  {
    const resource = setup();
    const { host } = resource;

    const app = createApp({
      template: '<b class="hello" style="color: red;">Hello World!!</b>',
    });

    app.mount(`#${HOST_ID}`);

    await assertSnapshot(t, host.innerHTML);
  }

  {
    const resource = setup();
    const { host } = resource;

    const app = createApp({
      template:
        '<p class="hello" label="Hello, ChibiVue!">Hello, ChibiVue!</p>',
    });

    app.mount(`#${HOST_ID}`);

    await assertSnapshot(t, host.innerHTML);
  }
});

Deno.test("変数を挿入できる", async (t) => {
  const resource = setup();
  const { host } = resource;

  const app = createApp({
    setup() {
      const state = reactive({ message: "Hello, ChibiVue!" });
      return { state };
    },
    template: '<p class="hello">{{ state.message }}</p>',
  });

  app.mount(`#${HOST_ID}`);

  await assertSnapshot(t, host.innerHTML);
});
