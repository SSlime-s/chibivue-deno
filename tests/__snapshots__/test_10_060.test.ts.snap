export const snapshot = {};

snapshot[`template もコンパイルできる 1`] = `'<b class="hello" style="color: red;">Hello World!!</b>'`;

snapshot[`template もコンパイルできる 2`] = `'<p class="hello" label="Hello, ChibiVue!">Hello, ChibiVue!</p>'`;

snapshot[`変数を挿入できる 1`] = `'<p class="hello">Hello, ChibiVue!</p>'`;
